packages/attestation-service/src/db.ts

```diff
@@ -53,38 +53,89 @@ export function isDBOnline() {
 }
 
 let kit: ContractKit | undefined
+let backupKit: ContractKit | undefined
+
+async function execWithFallback<T>(
+  f: (kit: ContractKit) => T,
+  kit1: ContractKit,
+  kit2?: ContractKit | undefined,
+  smartFallback = true
+): Promise<T> {
+  // Decorator to wrap execution of f with a retry using the kit2 kit
+  // if execution initially fails
+
+  let primaryKit = kit1
+  let secondaryKit = kit2
+  if (
+    smartFallback &&
+    kit2 &&
+    // TODO: look into whether we need to split this up out of the if statement
+    // check if not undefined, and have extra handling for that (i.e. take the other)
+
+    // Prioritize block age over syncing/not; avoids complex checks for stalls
+    // and assumes that if one is stalled, the next block number check will correct this.
+    (!kit1 ||
+      (await getAgeOfLatestBlockFromKit(kit2)).number >
```

Comment:
One concern I have is that if an endpoint is down or has high latency, then we slow down every call with this preliminary check. To address this we can add a timeout and error handling to `getAgeOfLatestBlockFromKit` to set an upper bound.

Reply:
totally -- think the timeout and error handling makes sense in any case. This may also be an argument in favor of decoupling the endpoint checks from the actual request servicing (i.e. Option 3) in that it wouldn't directly slow processing of the call, would be interested to hear your thoughts on this later!

Reply:
resolved with #8449 (rebased and merged into this branch)